### ISSUE TITLE
Mockup user profile editing UI?

### ISSUE THREAD

**Issue by bfackland** [ploneintranet/ploneintranet.prototype#61]
@cornae @mattss The user profile editing UI will need to be considered in the near future; specifically because some fields will be editable by the user and other fields will often be managed centrally by HR or IT. Rather than just hiding the non-editable fields I would prefer for us to have some text that states:

"If you wish to update your [First Name], [Last Name] etc. please contact [link to user or name of department]." This could just be a simple banner at the top of the page I guess.

@gyst My question at this stage is: do we intend to mockup the editing UI for user profiles?

**Comment by cornae** [comment]
I'm working on a form that contains a lot fields that the user may or may not edit that are not directly visible on the profile landing page. I'm considering to put this form under a tab (info or similar) on the profile section.

**Comment by bfackland** [comment]
Great!

**Comment by mattss** [comment]
@cornae When viewing my own profile there is an 'info' tab which shows some properties in editable fields, but there doesn't seem to be any 'save' button or similar. Is the info tab the correct way for the user to edit their profile, and if so how do they save changes?

**Comment by cornae** [comment]
That is correct, but I probably forgot to add pat-autosubmit + feedback page with a self healing message it?

**Comment by mattss** [comment]
@cornae Is pat-autosubmit the best way forward with forms that actually change content? I can't think of anywhere else in the prototype where that happens; auto-submit is usually used for getting/searching content when there is clear visual feedback. My concern is that auto-submitting this page would result in accidental data changes.

**Comment by cornae** [comment]
It’s a giant debate in UX land, accidental data changes only one of the many cons against the many pros, which I don’t want ignite here right now ;). There’s IMO not one truth, but there is however a strong trend towards auto save on web apps. An example of a major web app is AirBnB, which removed all the save buttons. Another one is Google docs (which is technically something different because it’s realtime collab, but that doesn’t matter for the user’s experience). So save buttons slowly disappear. Settings screens on a Mac have never had save buttons, unlike Windows which had always make and even an in between state ‘apply’. On the web, different rules now apply.

The fact that there is a save button for our rich text pages is because of a technical constraint of the moment. I don’t really want one there either, so here, and also on the settings section of a workspace, there all is auto save. Exceptions do apply for dialogues like change password or e-mail address or some other mutations with possibly large consequences. 

My personal concern is more that the self healing messages are a bit too much in the face for this purpose and I want to introduce a more subtel variant for autosave in the future.

**Comment by mattss** [comment]
@cornae hehe OK sorry for opening that can of worms(!)
